core/src/alloc.rs

```diff
@@ -1,13 +1,22 @@
+#[cfg(not(feature = "custom_allocator"))]
```

Comment:
Maybe a separate module `heap` could help reducing the number of feature flags. The module could be `pub` while the alloc module remains private.

Reply:
Good idea! I added a `heap` module enabled by a `heap` feature. The `alloc` feature then depends on `heap`.